File: Python/Algorithm/baekjoon/2559_수열.py
# baekjoon source = "https://www.acmicpc.net/problem/2559"
'''
10 2
3 -2 -4 -9 0 3 7 13 8 -3

10 5
3 -2 -4 -9 0 3 7 13 8 -3

5 3
5 5 5 2 1000

100000 5
'''
import sys

N, K = map(int,sys.stdin.readline().split())
num = list(map(int,sys.stdin.readline().split()))

max_ = 0; sum_ = 0
for i in range(K):
    sum_ += num[i]
    max_ = sum_

for i in range(K,N):
    sum_ -= num[i-K]
    sum_ += num[i]
    max_ = max(max_,sum_)

print(max_)





# import sys
# import time

# st = time.time()

# N, K = map(int,sys.stdin.readline().split())
# # num = list(map(int,sys.stdin.readline().split()))
# num = list(range(N))
# numbers = result = 0

# for i in range(N-K+1):
#     numbers += sum(num[i:i+K])
#     result = max(result,numbers)
#     numbers = 0

# print(result)

# print((time.time() - st))

File: Java/baekjoon/IM/src/_2559_수열.java

//baekjoon source = "https://www.acmicpc.net/problem/2559"

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class _2559_¼ö¿­ {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int k = Integer.parseInt(st.nextToken());
		int[] arr= new int[n];
		st = new StringTokenizer(br.readLine());
		for(int i = 0; i<n; ++i) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		
		int sum = 0, result = 0;
		for(int i = 0; i<k; ++i) {
			sum += arr[i];
		}
		result = sum;
		
		for(int i = 0; i<n-k; ++i) {
			sum -= arr[i];
			sum += arr[i+k];
			result = Math.max(result, sum);
		}
		bw.write(result+"");
		bw.flush();
	}
}
